### ISSUE TITLE
Expression compilation fails for generic methods 

### ISSUE THREAD

**Issue by aristovg** [ExRam/ExRam.Gremlinq#18]
Hi, 
First of all, thanks for the great work!

I was trying to use some generic methods limited by constraints, but it seems that the library is not yet able to cope with them. 

A minimal example that reproduces the error looks like 

```cs
public interface INamed
{
    string Name { get; set; }
}

public class NamedPerson : Vertex, INamed
{
    public string Name { get; set; }
}

public class NamedAnimal : Vertex, INamed
{
    public string Name { get; set; }
}


[Fact]
public void Explicit()
{
    //this passes as expected
    g.V<INamed>()
        .Where(v => v.Name == "example")
        .Should()
        .SerializeToGroovy<CosmosDbGroovyGremlinQueryElementVisitor>("g.V().hasLabel(_a, _b).has(_c, _d)")
        .WithParameters("NamedAnimal", "NamedPerson", "Name", "example");
}

[Fact]
public void Generic()
{
    //ExRam.Gremlinq.Core.ExpressionNotSupportedException : An expression is not supported.
    //at ExRam.Gremlinq.Core.GremlinQuery`6.Where
    FindByName<NamedPerson>(g)
        .Should()
        .SerializeToGroovy<CosmosDbGroovyGremlinQueryElementVisitor>("g.V().hasLabel(_a).has(_b, _c")
        .WithParameters("NamedPerson", "Name", "example");
}

public IVertexGremlinQuery<TNamed> FindByName<TNamed>(IConfigurableGremlinQuerySource source) where TNamed : INamed
{
    return source.V<TNamed>().Where(x => x.Name == "example");
}
```

It seems that compiler inserts _Convert_ expression under the hood, so the Where method predicate becomes _{x => (Convert(x, INamed).Name == "example")}_
Unfortunately, that throws ExRam.Gremlinq.Core.ExpressionNotSupportedException. 

I'm not sure that's it possible to work around this issue, so the best solution that I found is just not using such generic methods. However, it would be nice to have support for it. 

Please let me know if this is on your radar, and if not - I'll try to take a more detailed look. 

Thanks!

**Comment by danielcweber** [comment]
Interesting scenario. Usually. those conversion should be stripped from the expressions. I'll have a look into it. If you'd like to try and tackle it yourself, make sure you go from the commit tagged v6.6.1.

**Comment by danielcweber** [comment]
Got it already. There'll be a fix.

**Comment by danielcweber** [comment]
The fix is part of the new Gremlinq 7.0.0 preview.
